src/main/java/io/jenkins/plugins/conventionalcommits/utils/GetCurrentVersion.java

```diff
@@ -0,0 +1,53 @@
+package io.jenkins.plugins.conventionalcommits.utils;
+
+import com.github.zafarkhaja.semver.Version;
+import java.lang.ProcessBuilder;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.io.IOUtils;
+
+public class GetCurrentVersion {
+
+
+    private Version getCurrentVersionMaven(File dir) throws IOException, InterruptedException {
+
+        String os = System.getProperty("os.name");
+        String commandName = "mvn";
+
+        if (os.contains("Windows")) {
+            commandName += ".cmd";
+        }
+
+        ProcessBuilder processBuilder = new ProcessBuilder(
+                commandName, "help:evaluate",
+                "-Dexpression=project.version", "-q", "-DforceStdout"
+        );
+
+        processBuilder.directory(dir);
+        Process process = processBuilder.start();
+
+        String results = IOUtils.toString(process.getInputStream(), StandardCharsets.UTF_8);
+        process.waitFor();
+
+        return Version.valueOf(results);
+    }
+
+    private Version getCurrentVersionTag(String latestTag){
+        return Version.valueOf(latestTag.isEmpty() ? "0.0.0" : latestTag);
+    }
+
+    public Version getCurrentVersion(File directory, String latestTag) throws IOException, InterruptedException {
+
+        GetProjectType projectType = new GetProjectType();
+        if (projectType.checkMaven(directory)){
+            return getCurrentVersionMaven(directory);
+        }
+        // To loop over supported types to check which type of project it is
+        // Return "current" version once value is true...
+
+        return getCurrentVersionTag(latestTag);
+    }
```

Comment:
I made a longer suggestion in the review comments about using the factory pattern.  That would make it really easy to switch this code to something similar to this (not exactly depending on what you do for implementation. testing for types, etc)

```suggestion
    public Version getCurrentVersion(File directory, String latestTag) throws IOException, InterruptedException {

        // ProjectTypeFactory will contain 1 static method which will return a new correct instance of the ProjectType class
        ProjectType projectType = ProjectTypeFactory.getProjectFromDirectory(directory); //all the loop logic to pick the right project type is in the getProjectFromDirectory method
        String latestTag =  projectType.getCurrentVersion(directory);
        return getCurrentVersionTag(latestTag);
    }
```

Reply:
You would of course need to create an Interface or AbstractClass `ProjectType` with a method `getCurrentVersion` that something like `MavenProjectType` would inherit/extend.

Eventually other project types like `HelmProjectType` or `GradleProjectType` would exist as well.